runtimes/runtimes/standalone.ts

```diff
@@ -92,7 +93,7 @@ if (checkAWSConfigFile()) {
  * @param props.servers The list of servers to initialize and run
  * @returns
  */
-export const standalone = (props: RuntimeProps) => {
+export const standalone = async (props: RuntimeProps) => {
```

Comment:
We should avoid changing signature of `standalone` function, as it is public interface for all consuming packages - we have many build configs today, and it is in a way breaking change.

Reply:
I see that this change is caused by `getLoggingUtility` requesting configuration from client in `getLogLevel()` with `lspConnection.workspace.getConfiguration`.

I think this is not correct approach, as it breaks LSP specification - it requests something by server from client before server even starts listening on the connection, and before server responds with `Initialize`. Were you able to test that it actually works with compiled server and real client? I suspect this may not work correctly.

https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initialize

> Until the server has responded to the initialize request with an [InitializeResult](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initializeResult), the client must not send any additional requests or notifications to the server. In addition **the server is not allowed to send any requests or notifications to the client until it has responded with an [InitializeResult](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#initializeResult)**, with the exception that during the initialize request the server is allowed to send the notifications window/showMessage, window/logMessage and [telemetry/event](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#telemetry_event) as well as the window/showMessageRequest request to the client.


Better solution will be:

* Accept `logLevel` option in [InitializeParams](https://github.com/aws/language-server-runtimes/blob/f6be37f05713ec5d1fc8156c8567f9e948b3894b/runtimes/protocol/lsp.ts#L115-L116).initializationOptions. Set reasonable default value, when not set (`log` or `info`)
* In runtime, subscribe to [workspace/didChangeConfiguration](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeConfiguration) notification from client, and only then pull and update config with `getLogLevel()`.

Reply:
I added the test cases but did not try out testing with the real client though. 

>it requests something by server from client before server even starts listening on the connection, and before server responds with Initialize
I see lspConnection.listen is at the end of the file. 

If I go ahead with **workspace/didChangeConfiguration**, then a part of the runtime will be initialized later while the other components would have been initialized early. Or you are suggesting to start with initializing logging as default log/info log level and then listen to configuration update and then update the logging level to a new value?

Reply:
I would prefer to go with workspace/didChangeConfiguration (if my understanding is correct as described in the above comment) over InitializeParams as it's more dynamic and we don't need to update the structure of initializeOptions as well.

Reply:
A doubt though: I think with didChangeConfiguration one as well, we already return the logging to all servers before while iterating them. Don't know if updating logging later would help here.

Reply:
We need both to support this feature fully: optional initial `logLevel` value passed in `initializationOptions` and subscription to `workspace/didChangeConfiguration`. 
Without `initializationOptions` server will always default to some value, while clients may be able to preset `logLevel` from the very start